eth/api_backend.go

```diff
@@ -59,13 +59,11 @@ func (b *EthAPIBackend) SetHead(number uint64) {
 }
 
 func (b *EthAPIBackend) HeaderByNumber(ctx context.Context, number rpc.BlockNumber) (*types.Header, error) {
-	// Pending block is only known by the miner
-	if number == rpc.PendingBlockNumber {
-		block := b.eth.miner.PendingBlock()
-		return block.Header(), nil
-	}
-	// Otherwise resolve and return the block
-	if number == rpc.LatestBlockNumber {
+	// Use latest block for validators, otherwise use pending block for full nodes
+	if number == rpc.LatestBlockNumber || number == rpc.PendingBlockNumber {
+		if !b.eth.IsMining() {
+			return b.eth.miner.PendingBlock().Header(), nil
+		}
```

Comment:
```suggestion
	// If this is a validator, return the latest block if pending block is requested.  Non proposing validators for IBFT don't have a pending block.  TODO(kevjue) modify the miner so that the pending txns will build up the pending block (similar to what full nodes do).
	if number == rpc.PendingBlockNumber {
		if !b.eth.IsMining() {   // This is a full node
			return b.eth.miner.PendingBlock().Header(), nil
		} else {
		        return b.eth.blockchain.CurrentBlock().Header(), nil
		}
	}
	
	// Otherwise resolve and return the block
	if number == rpc.LatestBlockNumber {
```